scipy/spatial/distance.py

```diff
@@ -2641,14 +2657,7 @@ def cdist(XA, XB, metric='euclidean', *args, **kwargs):
     elif isinstance(metric, string_types):
         mstr = metric.lower()
 
-        # NOTE: C-version still does not support weights
-        if "w" in kwargs and not mstr.startswith("test_"):
-            if(mstr in _METRICS['seuclidean'].aka or
-               mstr in _METRICS['mahalanobis'].aka):
-                raise ValueError("metric %s incompatible with weights" % mstr)
-            # need to use python version for weighting
-            kwargs['out'] = out
-            mstr = "test_%s" % mstr
+        mstr, kwargs = _select_weighted_metric(mstr, kwargs, out)
```

Comment:
To further reduce duplication, include the `mstr.lower()` call in `_select_weighted_metric`.

Reply:
I'd like to keep it here for readability, since the fact that it's lowercased is used below.

Anyway, all this code could use refactoring and simplification, it's a bit complex just for argument parsing for my taste.